src/TelnetClient.php

```diff
@@ -238,6 +253,15 @@ protected function getResponse($prompt = null, $promptError = null)
                 $isError = true;
                 break;
             }
+
+            // throw an exception if the number of characters read is greater than the limit
+            if ($this->maxReadCharacters > 0 && strlen($buffer) >= $this->maxReadCharacters) {
```

Comment:
This is a micro-optimisation, but consider just keeping a counter rather than calling `strlen()` for every byte read, given the value is only ever going to increment by 1 anyway.

Reply:
Every little helps 87b9201